hikari/impl/stateful_cache.py

```diff
@@ -1415,19 +1445,13 @@ def clear_members_for_guild(
         if guild_record is None or guild_record.members is None:
             return _EmptyCacheView()
 
-        cached_users = {user_id: self._user_entries[user_id] for user_id in guild_record.members}
-        cached_members = {
-            member.id: member
-            for member in itertools.starmap(
-                self._chainable_remove_member, ((m, guild_record) for m in tuple(guild_record.members.values())),
-            )
-            if member is not None
-        }
-
-        if not guild_record.members:
-            guild_record.members = None
-            self._remove_guild_record_if_empty(guild_id)
-
+        cached_members = _copy_mapping(guild_record.members)
+        # see https://github.com/python/mypy/issues/4297 for type ignore
+        members_map = map(
```

Comment:
So map is faster than generator expressions?

Reply:
It's not quite as black and white as that and while a map along will be faster than a generator expression, introducing lambdas slows stuff down a considerable amount time (although if you're passing enough constant args the map will end up overtaking the generator expression but that's such a deep rabbit hole I don't want to think about it).
But the use of map specifically here is more an artefact of how i was doing stuff before so I'll probably switch that to a generator expression.